src/bindfs.c

```diff
@@ -904,10 +906,37 @@ static int bindfs_mknod(const char *path, mode_t mode, dev_t rdev)
 
     mode = permchain_apply(settings.create_permchain, mode);
 
-    if (S_ISFIFO(mode))
+    if (S_ISFIFO(mode)) {
         res = mkfifo(real_path, mode);
-    else
+#ifdef __APPLE__
+    } else if (S_ISSOCK(mode)) {
+        struct sockaddr_un su;
+        int fd;
+
+        if (strlen(real_path) >= sizeof(su.sun_path)) {
+            errno = ENAMETOOLONG;
+            return -1;
+        }
+        fd = socket(AF_UNIX, SOCK_STREAM, 0);
+        if (fd >= 0) {
+            /*
+             * We must bind the socket to the underlying file
+             * system to create the socket file, even though
+             * we'll never listen on this socket.
+             */
+            su.sun_family = AF_UNIX;
+            strncpy(su.sun_path, real_path, sizeof(su.sun_path));
+            res = bind(fd, (struct sockaddr*)&su, sizeof(su));
+            if (res == 0)
```

Comment:
Don't we need to close regardless of `bind`'s success?

Reply:
I don't think so. `nc` wants to read from it. I'll test what happens in case we close it later today. As I understand, you're worried that file descriptor will leak in case listening process dies? Need to think about it... I'll also take a look at how you handle the same issue with `fopen`.

Reply:
Oh, we don't return `fd` anywhere. Maybe you're right. Will test and update accordingly.

Reply:
I drank some more coffee, you're right, we're just leaking socket fd into the void.

Reply:
I've tested, everything works properly with unconditional socket closing. Updated PR.